### ISSUE TITLE
Unlocking Shared Fails after osrm-routed crashes

### ISSUE THREAD

**Issue by MoKob** [Project-OSRM/osrm-backend#3574]
While running local experiments on a crashing request, I am not able to re-instate OSRM into a sane state.

```
./build/osrm-datastore data/berlin.osrm
[warn] Data update in progress, waiting until it finishes... 

./build/osrm-unlock-all 
[info] Releasing all locks

./build/osrm-springclean 
[info] Releasing all locks
[info] ATTENTION! BE CAREFUL!
[info] ----------------------
[info] This tool may put osrm-routed into an undefined state!
[info] Type 'Y' to acknowledge that you know what your are doing.
[info] 

Do you want to purge all shared memory allocated by osrm-datastore? [type 'Y' to confirm]
Y
[info] spring-cleaning all shared memory regions

./build/osrm-datastore data/berlin.osrm
[warn] Data update in progress, waiting until it finishes... 
```

It seems that unlocking shared memory is not possible with our toolchain anymore.

Edit:
a previous run of `osrm-springclean` showed:
```
[debug] deallocating prev memory 28049409
[debug] deallocating prev memory 27983872
[debug] deallocating prev memory 65538
```

Edit 2:
Even using `max-wait` does not work:

```
./osrm-datastore --max-wait 1 ../data/berlin.osrm
[warn] Data update in progress, waiting until it finishes... 
```
hangs for minutes.

Edit 3:
I got to this state running `osrm-datastore OSRM-FILE`, ran `osrm-routed -s` and had it crash multiple times. After I cannot run any queries anymore against osrm.

**Comment by oxidase** [comment]
@MoKob please could you check `./osrm-datastore --remove-locks` that is similar to `./osrm-unlock-all`.
`--max-wait` options now is used only to get write access to `CURRENT_REGION` block

**Comment by MoKob** [comment]
@oxidase thank you for the tip. I will see if I can reproduce it again, since I kind of needed my PC and had to restart.

Regardless of the outcome, I feel that we should opt to have a clear way of disabling locks. Right now I can see 4 ways to possibly remove locks, if everything hangs. This is kind of confusing and does not make it easy to find the source of the problem.

**Comment by oxidase** [comment]
@MoKob i have created #3577 to reduce confusion in tools.

Please check
```
I got to this state running osrm-datastore OSRM-FILE, ran osrm-routed -s and had it crash multiple times. After I cannot run any queries anymore against osrm.
```
If it is still reproduced then more context is needed, it could be an OSX related issue

**Comment by oxidase** [comment]
@MoKob please check if it is an issue. `osrm-datastore --remove-locks` and `rm /tmp/osrm-datastore.lock` should clean completely the locking state  (removing /tmp/osrm-datastore.lock is not really needed because only inode is used for fcntl, killing locked `osrm-datastore` is enough)

**Comment by MoKob** [comment]
@oxidase I did some checks to reproduce.

In [here](https://github.com/Project-OSRM/osrm-backend/blob/c277b95f039dd42153f540dc6552d7ef3d0faf7a/src/engine/api/json_factory.cpp#L186), I made the assertion fail always. (`assert(false)`);

After I ran `osrm-datastore file` and `osrm-route -s`.
After two crashes, osrm-routed does not start anymore and `osrm-datastore` the same.

With the `osrm-datastore --remove-locks`, everything can be restored to normal again.
